### PR TITLE
chore: Introduce splunk_hec -> route -> s3 soak

### PR DESCRIPTION
**Pull request by blt**
This commit introduces a soak for the above configuration. Interestingly vector
crashes owing to being too memory hungry and the soak does not run as a result.

REF #9515 

Signed-off-by: Brian L. Troutwine <brian@troutwine.us>

### CODE REVIEW

File: soaks/tests/splunk_hec_route_s3/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Splunk HEC -> Route Transform -> s3
```
Comment (jszwedko) [inline soaks/tests/splunk_hec_route_s3/README.md:1]: Just a note that I noticed we stopped updating the index of soaks in `soaks/README.md`. I think that's fine and would prefer to just see one of these README files for each soak test.

Comment (blt) [reply soaks/tests/splunk_hec_route_s3/README.md:1]: Tell you what, I fully forgot about that index. If you wanted to make a PR to remove that I'd be for it. 

I'm also happy to nuke it later.

**Comment by blt** [comment]
`splunk_hec_route_s3` is crashing because it's running out of memory. Quick note here that our CI machines have more headroom than local runs for memory and we can bump the limits in CI.

File: soaks/tests/splunk_hec_route_s3/terraform/http_gen.toml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+[targets.vector]
+target_uri = "http://vector:8282/services/collector/event/1.0"
+bytes_per_second = "120 Mb"
+parallel_connections = 50
+method.type = "Post"
+method.variant = "SplunkHec"
+method.maximum_prebuild_cache_size_bytes = "256 Mb"
+[targets.vector.headers]
+dd-api-key = "DEADBEEF"
```
Comment (spencergilbert) [inline soaks/tests/splunk_hec_route_s3/terraform/http_gen.toml:12]: What is the origin behind DEADBEEF

**Review by spencergilbert — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]
Nice! The implementation looks good. The numbers look pretty bad though, are they what you expected?

**Comment by blt** [comment]
> Nice! The implementation looks good. The numbers look pretty bad though, are they what you expected?

The numbers mean the test crashed while running, so 24g wasn't enough memory for the config.

File: soaks/tests/splunk_hec_route_s3/terraform/http_gen.toml (hunk 2 above)
Comment (blt) [reply soaks/tests/splunk_hec_route_s3/terraform/http_gen.toml:12]: It's just a hex word.

**Comment by jszwedko** [comment]
Nice! Those numbers are looking better. I know that soak test has a lot of batch partitions so requires a substantial amount of memory.

**Comment by blt** [comment]
> Nice! Those numbers are looking better. I know that soak test has a lot of batch partitions so requires a substantial amount of memory.

Yep. The runners have 32Gb of RAM available and now vector has access -- through the minikube -- to 30 of it.

**Comment by blt** [comment]
Missing fields to be addressed in #9973

**Comment by spencergilbert** [comment]
That's a hungry hungry setup

**Comment by blt** [comment]
> That's a hungry hungry setup

Indeed. It'll be interesting to tackle https://github.com/vectordotdev/vector/issues/9946 and measure how hungry it is.

**Pull request merged**